data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/ResponseExhaustJobV2.scala

```diff
@@ -0,0 +1,104 @@
+
+  override def processBatch(userEnrolmentDF: DataFrame, collectionBatch: CollectionBatch)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): DataFrame = {
+    val assessmentDF = getAssessmentDF(userEnrolmentDF, collectionBatch).persist();
+    persistedDF.append(assessmentDF);
+    val contentIds = assessmentDF.select("content_id").dropDuplicates().collect().map(f => f.get(0));
+    val contentDF = searchContent(Map("request" -> Map("filters" -> Map("identifier" -> contentIds)))).withColumnRenamed("collectionName", "contentname").select("identifier", "contentname");
+    val reportDF = assessmentDF.join(contentDF, assessmentDF("content_id") === contentDF("identifier"), "left_outer").drop("identifier").select(filterColumns.head, filterColumns.tail: _*);
+    organizeDF(reportDF, columnMapping, columnsOrder);
+  }
+
+  override def unpersistDFs() {
+    persistedDF.foreach(f => f.unpersist(true))
+  }
+
+  def getAssessmentDF(userEnrolmentDF: DataFrame, batch: CollectionBatch)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): DataFrame = {
+
+    val userEnrolmentDataDF = userEnrolmentDF
+      .select(
+        col("userid"),
+        col("courseid"),
+        col("collectionName"),
+        col("batchName"),
+        col("batchid"))
+
+    val assessAggregateData = loadData(assessmentAggDBSettings, cassandraFormat, new StructType()).where(col("batch_id").equalTo(batch.batchId))
+
+    val preparedAssessAggregateData = prepareReportDf(assessAggregateData)
+
+    val joinedDF = try {
+      val assessBlobData = prepareReportDf(getAssessmentBlobDF(batch, config))
+      preparedAssessAggregateData.unionByName(assessBlobData).dropDuplicates("course_id","batch_id", "user_id", "attempt_id", "questionid")
+    } catch {
+      case e: Exception => {
+        JobLogger.log("Blob does not contain any file for batchid: " + batch.batchId)
+        preparedAssessAggregateData
+      }
+    }
+
+    joinedDF.join(userEnrolmentDataDF, joinedDF.col("user_id") === userEnrolmentDataDF.col("userid") && joinedDF.col("course_id") === userEnrolmentDataDF.col("courseid") && joinedDF.col("batch_id") === userEnrolmentDataDF.col("batchid"), "inner")
+  }
+
+  def getAssessmentBlobDF(batch: CollectionBatch, config: JobConfig)(implicit spark: SparkSession, fc: FrameworkContext): DataFrame = {
+    val azureFetcherConfig = config.modelParams.get("assessmentFetcherConfig").asInstanceOf[Map[String, AnyRef]]
+
+    val store = azureFetcherConfig("store").asInstanceOf[String]
+    val format:String = azureFetcherConfig.getOrElse("format", "csv").asInstanceOf[String]
+    val filePath = azureFetcherConfig.getOrElse("filePath", "archival-data/").asInstanceOf[String]
+    val container = azureFetcherConfig.getOrElse("container", "reports").asInstanceOf[String]
```

Comment:
Probably this container may required to change. Will check and update in the job config.